case_studies/galaxy_clustering/prior.py

```diff
@@ -0,0 +1,338 @@
+        self.cluster_max_z = 0.5
+        self.G1_beta = 0.6
+        self.G1_loc = 0
+        self.G1_scale = 0.035
+        self.G2_beta = 0.6
+        self.G2_loc = 0
+        self.G2_scale = 0.032
+        self.tiles_width = 25
+        self.tiles_height = 25
+        with open("gal_gmm_nmgy.pkl", "rb") as f:
+            self.gmm_gal = pickle.load(f)
+        self.tsize_poly = np.poly1d([-6.88890387e-11, 3.70584026e-5, 4.34623392e-2])
+        self.folder_path = "data5/"
+        self.threadings = 8
+        self.redshift_alpha = 1.24
+        self.redshift_beta = 1.01
+        self.redshift0 = 0.51
+        self.cluster_prob = 0.2
+
+    #  Source: https://github.com/LSSTDESC/CLMM/blob/main/clmm/redshift/distributions.py
+    def _redshift_distribution(self, redshift):
+        return (redshift**self.redshift_alpha) * np.exp(
+            -((redshift / self.redshift0) ** self.redshift_beta)
+        )
+
+    def _toflux(self, mag):
+        return 10 ** ((mag - 30) / -2.5)
+
+    def _cluster_threshold(self, r):
+        return 3 / (2 * np.pi * (r**2))
+
+    def _sample_mass(self):
+        hmf = MassFunction()
+        hmf.update(Mmin=14.5, Mmax=15.5)
+        mass_func = hmf.dndlnm
+        mass_sample = []
+        while len(mass_sample) < self.size:
+            index = np.random.randint(0, len(mass_func))
+            prob = (mass_func / sum(mass_func))[index]
+            if np.random.random() < prob:
+                mass_sample.append(
+                    (self.mass_max - self.mass_min) / len(mass_func) * (index + np.random.random())
+                    + self.mass_min,
+                )
+        return mass_sample
+
+    def z_to_zis(self, z, mass, size):
+        delta = (WMAP7.H(z).value / 100 * mass / (10**15 * 1.989 * 10**33)) ** (1 / 3) * 1082.9
+        # light speed
+        c = 899377.37
+        speed_mean = (c * (1 + z) ** 2 - c) / ((1 + z) ** 2 + 1)
+        zi = np.random.normal(speed_mean, delta, size)
+        return [max(0, np.sqrt((c + x) / (c - x)) - 1) for x in zi]
+
+    def z_m_to_r(self, mass, z):
+        pho_z = WMAP7.critical_density(z).value
+        return (mass / (4 / 3 * np.pi * pho_z)) ** (1 / 3)
+
+    def _sample_redshift(self):
+        redshift_bins = np.linspace(0.01, 7, 100)
+        redshift_pdf = [self._redshift_distribution(z) for z in redshift_bins]
+        redshift_pdf = redshift_pdf / np.sum(redshift_pdf)
+        redshift_samples = np.random.choice(
+            np.linspace(0.01, 7, 100), size=self.size, p=redshift_pdf
+        )
+        for i in range(self.size):
+            redshift_samples[i] += (np.random.random()) * 0.07
+        return redshift_samples
+
+    def _sample_radius(self, mass_samples, redshift_samples):
+        radius_samples = []
+        for i in range(self.size):
+            radius_samples.append(
+                self.z_m_to_r(mass_samples[i], redshift_samples[i])
+                / (3.086 * 10**24)
+                * self.scale_pixels_per_au,
```

Comment:
I'd be clearer if we used some intermediate variables, rather than have a multi-line append statement